x-pack/platform/plugins/shared/fleet/server/services/agent_policies/full_agent_policy.ts

```diff
@@ -893,23 +893,255 @@ export function getBinarySourceSettings(
+  });
+
+  return { [type]: generated };
+}
+
+function addSuffixToOtelcolPipelinesComponents(pipelines: any, suffix: string) {
+  const result: Record<string, any> = {};
+  Object.entries(pipelines as Record<string, Record<string, string[]>>).forEach(
+    ([pipelineID, pipeline]) => {
+      const newPipeline: Record<string, any> = {};
+      Object.entries(pipeline).forEach(([type, componentIDs]) => {
+        newPipeline[type] = componentIDs.map((id) => id + '/' + suffix);
+      });
+      result[pipelineID] = newPipeline;
+    }
+  );
+  return result;
+}
+
+function mergeOtelcolConfigs(otelConfigs: OTelCollectorConfig[]) {
+  return otelConfigs.reduce((merged, next) => {
+    if (!next) {
+      return merged;
+    }
+    const extensions = {
+      ...merged.extensions,
+      ...next.extensions,
+    };
+    const receivers = {
+      ...merged.receivers,
+      ...next.receivers,
+    };
+    const processors = {
+      ...merged.processors,
+      ...next.processors,
+    };
+    const connectors = {
+      ...merged.connectors,
+      ...next.connectors,
+    };
+    const exporters = {
+      ...merged.exporters,
+      ...next.exporters,
+    };
+    return {
+      ...(Object.keys(extensions).length > 0 ? { extensions } : {}),
+      ...(Object.keys(receivers).length > 0 ? { receivers } : {}),
+      ...(Object.keys(processors).length > 0 ? { processors } : {}),
+      ...(Object.keys(connectors).length > 0 ? { connectors } : {}),
+      ...(Object.keys(exporters).length > 0 ? { exporters } : {}),
+      service: {
+        ...merged.service,
+        ...(next.service?.extensions
+          ? {
+              extensions: (merged.service?.extensions ? merged.service.extensions : []).concat(
+                next.service.extensions
+              ),
+            }
+          : {}),
+        pipelines: {
+          ...merged.service?.pipelines,
+          ...next.service?.pipelines,
+        },
+      },
+    };
+  });
+}
+
+function attachExporter(config: OTelCollectorConfig, dataOutput?: Output) {
+  if (!dataOutput) {
+    return config;
+  }
+
+  const exporter = generateExporter(dataOutput);
+  config.connectors = {
+    ...config.connectors,
+    forward: {},
+  };
+  config.exporters = {
+    ...config.exporters,
+    ...exporter,
+  };
+
+  if (config.service?.pipelines) {
+    const signalTypes = new Set<string>();
+    Object.entries(config.service.pipelines).forEach(([id, pipeline]) => {
+      config.service!.pipelines![id] = {
+        ...pipeline,
+        exporters: [...(pipeline.exporters || []), 'forward'],
+      };
+      signalTypes.add(signalType(id));
+    });
+
+    signalTypes.forEach((id) => {
+      config.service!.pipelines![id] = {
+        receivers: ['forward'],
+        exporters: Object.keys(exporter),
+      };
+    });
+  }
+
+  return config;
+}
+
+function generateExporter(dataOutput: Output) {
+  switch (dataOutput.type) {
+    case outputType.Elasticsearch:
```

Comment:
In principle supporting remote Elasticsearch would be as easy as adding it here. Though I haven't tested it yet.
```suggestion
    case outputType.Elasticsearch:
    case outputType.RemoteElasticsearch:
```